lib/strings.nix

```
@@ -678,4 +678,31 @@ rec {
        => "1.0"
   */
   fileContents = file: removeSuffix "\n" (builtins.readFile file);
+
+  /* Determines a valid nix store name for a path.
+
+     Type: stringToValidNixPath :: string -> string -> string
+
+     Example:
+       stringToValidNixPath "foo_" ".stack/config.yml"
+       => "foo_.stackconfig.yml"
+       stringToValidNixPath "  " ".stack/config.yml"
+       => "safePrefix_.stackconfig.yml"
+  */
+
+  stringToValidNixPath = prefix: path:
```

Does this sounds better?

```suggestion
  sanitizeNixPath = prefix: path:
```

---

That sounds much better!